numpy/core/tests/test_arrayprint.py

```diff
@@ -237,12 +243,26 @@ def test_formatter_reset(self):
         assert_equal(repr(x), "array([ 0.,  1.,  2.])")
 
     def test_0d_arrays(self):
+        if sys.version_info[0] >= 3:
+            assert_equal(str(np.array('café', np.unicode_)), 'café')
+            assert_equal(repr(np.array('café', np.unicode_)),
+                         "array('café',\n      dtype='<U4')")
```

Comment:
The newline appears for all flexible types. Probably this is because they can be very long. There is a special clause `if issubclass(arr.dtype.type, flexible):` in `array_repr` in `numeric.py` controlling it.